### DISCUSSION TITLE
RDP Session without username & password prompt

### DISCUSSION THREAD

**Discussion by marierau98** [FreeRDP/FreeRDP#7174] [Q&A] [answered]
Hello! 
Iam currently working on a new image with thinstation for our terminal pcs. I want them to connect to our rds with the known windows login screen but that isnt the case. When I start the image, my clients receive a prompt from freerdp to enter their username and password and only after entering them they are connected to the terminal server. Ive tried to disable nla with several commands like --sec-nla; -sec-nla etc but nothing is helping to resolve the issue. Is there another way or config lines I have to add to get it working?

Best regards

Marie

**Comment by akallabeth** [comment]
This is something the client can not force but that needs to be enabled on the server too.
You need a server that allows `TLS` instead of `NLA`.
The client options just enable/disable the path for the client but you will not get a login if the server has not enabled that feature.

**Reply by marierau98** [reply]
We already have thinclients working that way. they are directly connecting to the server and they only need to enter username and password via windows login screen, thats why iam thinking it must be a setting in my configuration.

**Comment by akallabeth** [comment, marked as answer]
the only options you have client side are `/sec:tls` (forces this to be the only allowed mode) or `-sec-nla` (disables `NLA`)

**Reply by marierau98** [reply]
Ive tried it and its still asking for a username beforehand :/

**Reply by akallabeth** [reply]
oh, you mean you´re struggling with the command line only?
did you try `/u: /p:` (empty user and password) to silence the command prompt?

**Reply by marierau98** [reply]
yes sorry if iam not clear enough! my parameters are: /cert-ignore /sec:tls /u: /p: /d:domainname /bpp:32 /rfx /gfx:AVC444 /sound:sys:pulse +auto-reconnect +heartbeat +fonts
but even if iam giving him a username and password at those u: and p: its still asking me for a username before it connects

**Reply by akallabeth** [reply]
which freerdp version are you using? tried this in several combinations with 2.3.2 and master, all of them work.

**Reply by marierau98** [reply]
iam using it with thinstation, i cant specify a version. is there a way to find it out?

**Reply by akallabeth** [reply]
ok, so you start it by script? maybe you need some quotes with your user/password things.

**Reply by marierau98** [reply]
I have a file with some parameters title, type, server ip, options and autostart. options are similar to the commands i would enter into my command line. if i use the exact same parameters from my "options" into my command line, it behaves as expected (connecting with empty username & password) but if i run my file it opens a white window with freerdp, username, password and its expecting me to enter a username, if i enter a space it does connect me to my server but the window still appears. my problem is on the thinstation side i suppose :/

**Reply by akallabeth** [reply]
yes. `xfreerdp` has no windows (except the session view window), all command line only.
